examples/ae_examples/cvae_examples/mlp_cvae_example/client.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+from pathlib import Path
+from typing import Sequence, Tuple
+
+import flwr as fl
+import torch
+import torch.nn as nn
+import torchvision.transforms as transforms
+from flwr.common.typing import Config
+from torch.nn.modules.loss import _Loss
+from torch.optim import Optimizer
+from torch.utils.data import DataLoader
+
+from examples.ae_examples.cvae_examples.mlp_cvae_example.models import MnistConditionalDecoder, MnistConditionalEncoder
+from fl4health.clients.basic_client import BasicClient
+from fl4health.model_bases.autoencoders_base import ConditionalVae
+from fl4health.preprocessing.autoencoders.loss import VaeLoss
+from fl4health.utils.dataset_converter import AutoEncoderDatasetConverter
+from fl4health.utils.load_data import load_mnist_data
+from fl4health.utils.metrics import Metric
+from fl4health.utils.random import set_all_random_seeds
+from fl4health.utils.sampler import DirichletLabelBasedSampler
+
+
+class CondAutoEncoderClient(BasicClient):
+    def __init__(
+        self, data_path: Path, metrics: Sequence[Metric], DEVICE: torch.device, condition: torch.Tensor
+    ) -> None:
+        BasicClient.__init__(self, data_path, metrics, DEVICE)
+        # In this example, condition is based on client ID.
+        self.condition_vector = condition
+        # To train an autoencoder-based model we need to define a data converter that prepares the data
+        # for self-supervised learning, concatenates the inputs and condition (packing) to let the data
+        # fit into the training pipeline, and unpacks the input from condition for the model inference.
+        # You can optionally pass a custom converter function to this class.
+        self.autoencoder_converter = AutoEncoderDatasetConverter(self.condition_vector)
+
+    def get_data_loaders(self, config: Config) -> Tuple[DataLoader, DataLoader]:
+        batch_size = self.narrow_config_type(config, "batch_size", int)
+        sampler = DirichletLabelBasedSampler(list(range(10)), sample_percentage=0.75, beta=100)
+        # ToTensor transform is used to make sure pixels stay in the range [0.0, 1.0].
+        # Flattening the image data to match the input shape of the model.
+        flatten_transform = transforms.Lambda(lambda x: torch.flatten(x))
+        transform = transforms.Compose([transforms.ToTensor(), flatten_transform])
+        train_loader, val_loader, _ = load_mnist_data(
+            data_dir=self.data_path,
+            batch_size=batch_size,
+            sampler=sampler,
+            transform=transform,
+            dataset_converter=self.autoencoder_converter,
+        )
+
+        return train_loader, val_loader
+
+    def get_criterion(self, config: Config) -> _Loss:
+        # The base_loss is the loss function used for comparing the original and generated image pixels.
+        # In this example, data is in binary scale, therefore binary cross entropy is used.
+        base_loss = torch.nn.BCELoss(reduction="sum")
```

Comment:
I have a bit of a naive question. Is the MNIST data binary? In my head I assumed it was "intensity." That is,  black and white, but a float [0, 1]. I might be wrong, but if they are intensity based, it think MSE makes more sense? Perhaps BCE loss also works though.

Reply:
Either way, I really like that we can strap on the VAE loss to "any" loss function.

Reply:
Thanks for catching that, it is not binary, so I will remove that comment. I think either can work, but I agree that MSE makes more sense.